test/new-e2e/agent-subcommands/subcommands_test.go

```diff
@@ -6,30 +6,214 @@
 package agentsubcommands
 
 import (
+	"fmt"
+	"regexp"
 	"testing"
 	"time"
 
 	"github.com/DataDog/datadog-agent/test/new-e2e/utils/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/utils/e2e/params"
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
 
 	"github.com/cenkalti/backoff"
 	"github.com/stretchr/testify/assert"
 )
 
 type subcommandSuite struct {
-	e2e.Suite[e2e.AgentEnv]
+	e2e.Suite[e2e.FakeIntakeEnv]
 }
 
-func TestStatusSuite(t *testing.T) {
-	e2e.Run(t, &subcommandSuite{}, e2e.AgentStackDef(nil))
+func TestSubcommandSuite(t *testing.T) {
+	e2e.Run(t, &subcommandSuite{}, e2e.FakeIntakeStackDef(nil), params.WithDevMode())
 }
 
-// XXX: this test is expected to fail until 7.48 as a known status render errors has been fixed in #18123
-func (v *subcommandSuite) TestNoStatusRenderError() {
+// section contains the content status of a specific section (e.g. Forwarder)
+type section struct {
+	name    string
+	content string
+}
+
+// getStatusComponentContent returns the content of the `sectionName` section if it exists, or an error otherwise
+func getStatusComponentContent(statusOutput string, sectionName string) (*section, error) {
+	// A status section is composed of:
+	// * a name (e.g. 'Forwarder')
+	// * followed by lines whose first character is not a blank (e.g. '==========')
+	// * and then lines starting with blank characters (which is basically the content of the section)
+	linesStartingWithNonWhiteCharRegex := `(?m:^\S.*\n)+`
+	linesStartingWithWhiteCharRegex := `(?P<status>(?m:^\s.*\n?)+)`
+	regexTemplate := fmt.Sprintf("%v\n", sectionName) + linesStartingWithNonWhiteCharRegex + linesStartingWithWhiteCharRegex
+
+	re := regexp.MustCompile(regexTemplate)
+	matches := re.FindStringSubmatch(statusOutput)
+
+	if len(matches) == 0 {
+		return nil, fmt.Errorf("regexp: no matches for %s status section", sectionName)
+	}
+
+	settingsIndex := re.SubexpIndex("status")
+
+	return &section{
+		name:    sectionName,
+		content: matches[settingsIndex],
+	}, nil
+}
+
+// expectedSection contains the information we want to verify about a section content
+type expectedSection struct {
+	name             string
+	shouldBePresent  bool
+	shouldContain    []string
+	shouldNotContain []string
+}
+
+func (v *subcommandSuite) TestDefaultInstallStatus() {
+
+	v.UpdateEnv(e2e.FakeIntakeStackDef(nil))
+	expectedSections := []expectedSection{
+		{
+			name:             `Agent \(.*\)`,
```

Comment:
Is it possible to get the version of the running agent from the stack and make sure it is the one which appears here ?

Reply:
Probably, I don't know if it can be done at the moment with the framework but I added a comment and I'll follow up